internal/reconcilers/hnc_config.go

```diff
@@ -186,14 +186,19 @@ func (r *ConfigReconciler) reconcileConfigTypes(inst *api.HNCConfiguration, allR
 		}
 
 		// Look if the resource exists in the API server.
-		gvk, err := GVKFor(gr, allRes)
```

Comment:
Would it be possible to use a custom error type, something like this?

```
gvk, err := GVKFor(gr, allRes)
if err != nil {
  reason := api.ReasonResourceNotFound
  if rle := err.(ResourceLookupError) { // or whatever the syntax is, I forget right now
     reason = rle.reason
  }
  ... everything else pretty obvious ...
}
```

That means we don't have to change the function signature or log/handle "errors" twice.

Reply:
finish it.